### PR TITLE
Prepare docker image for deployment using CircleCI

### PR DESCRIPTION
**Pull request by andyhd**
## What

* Added Dockerfile for API webapp
* Updated CircleCI config to build docker image and run tests using it, then push to quay.io, so that it can be deployed to the cluster

## How to review

1. See [CircleCI build logs](https://circleci.com/gh/ministryofjustice/analytics-platform-control-panel/75)
2. See [quay.io images](https://quay.io/repository/mojanalytics/control-panel?tab=tags)

See [#342 API and frontend deployment](https://trello.com/c/jxufKhNR)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
+FROM alpine:3.6
+
+MAINTAINER Andy Driver <andy.driver@digital.justice.gov.uk>
+
+RUN apk update && \
+    apk add \
+        --virtual build-deps \
+        gcc \
+        musl-dev \
+        postgresql-dev \
```
Comment (xoen) [inline Dockerfile]: Is this needed to install some of the requirements? Postgres should run in a separate container right? (using docker compose to run the tests).

**Comment by xoen** [comment]
As far as I can tell looks good

File: Dockerfile (hunk 1 above)
Comment (andyhd) [inline Dockerfile]: Yes, these are dependencies of `psycopg2` the postgres client library

Comment (xoen) [inline Dockerfile]: 👍

**Pull request merged**